### DISCUSSION TITLE
Rerunning Van Der Pol Tag

### DISCUSSION THREAD

**Discussion by dannyp3** [JuliaPOMDP/POMDPs.jl#560] [General]
I have a request about the results from the following paper: [Online algorithms for POMDPs with continuous state, action, and observation spaces](https://arxiv.org/abs/1709.06196)

I found the original code used to run the results from this paper here: [ContinuousPOMDPTreeSearchExperiments](https://github.com/zsunberg/ContinuousPOMDPTreeSearchExperiments.jl/tree/master?tab=readme-ov-file)

I am trying to rerun the Van Der Pol Tag POMDP problem with continuous state, action, and observation spaces referenced in this paper directly from the ContinuousPOMDPTreeSearchExperiments repository. I am hoping to solve it using the PFT-DPW solving method. So far, I have found that the ContinuousPOMDPTreeSearchExperiments repository mentions in the README that the results were run using Julia version 0.6. I have tried to rerun the results from the paper, but have not been able to run it through Julia. I am assuming this is because the code was written several years back with an outdated version of Julia

Also, I have noticed that there is an implementation of PFT-DPW referenced from the general JuliaPOMDPs repository page: [ParticleFilterTrees](https://github.com/WhiffleFish/ParticleFilterTrees.jl). It appears, though, that this repository was not used to generate the results from the paper I mentioned above.

Could I get some tips/help with rerunning the Van Der Pol Tag POMDP problem using the PFT-DPW solver? I am hoping to replicate the results from the paper highlighted in the following image:

![POMDP Image](https://github.com/user-attachments/assets/ef36aa1c-4442-428f-9bb0-75855bac628f)

Thank you so much!!

**Comment by zsunberg** [comment]
Hey @dannyp3 , thanks for reaching out and sorry for the delayed response - it is a busy time at the beginning of the semester.

Sorry that it is not too easy to reproduce the results because they are older and in Julia 0.6. It is probably easier to build off this paper: https://jair.org/index.php/jair/article/view/14525 which has experiment code here: https://github.com/WhiffleFish/PFTExperiments

I believe that uses the PFT-DPW implementation in ParticleFilterTrees and a Julia 1 compatible implementation of VDP Tag. I think that if you get the hyperparameters from there, you should be able to reproduce the result. Let us know how it goes! Hopefully @WhiffleFish can assist if you run into any problems.
